### DISCUSSION TITLE
I want to know how I can change the styling of the edge when onConnectStart gets triggered.

### DISCUSSION THREAD

**Discussion by surya0180** [xyflow/xyflow#3213] [Q&A] [answered]
Basically I have a design where the edges are styled based on the colors that are given to the nodes. Somehow I got access to the styles of the node. The issue now is that I want to give the edge the same color as my node and also give it a arrowhead. How can I achieve this when onConnectStart gets triggered?

**Comment by bcakmakoglu** [comment, marked as answer]
Well there is two things here - one is an *edge* which is the line between nodes and another is the *connection line* which is the line you draw *to connect two nodes*. When a connection is successful, an edge is created.

So your question is referring to edges but you're asking about the `onConnectStart` event so I'm not 100% sure which one you are trying to style as they are two different components.

[In any case here's a *simplified* example of how to style both of these based on some source node styles.](https://codesandbox.io/s/blue-cloud-ggr4pv?file=/src/Flow.tsx)

**Reply by surya0180** [reply]
Hey @bcakmakoglu! The connectionLineStyle is exactly what I am looking for. The codesandbox link is really helpful. Thanks a lot for the help.

I am closing this discussion now.
